### PR TITLE
Port generated hash functions

### PR DESCRIPTION
**Pull request by marcogschmidt**
Port of #2004 
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/1711

### CODE REVIEW

**Comment by marcogschmidt** [comment]
gRPC webtest flake /kick

**Comment by marcogschmidt** [comment]
Regression test flake /kick

**Comment by marcogschmidt** [comment]
Another gRPC web flake, apparently not resolved... /kick

File: projects/gloo/pkg/plugins/kubernetes/eds.go [hunk 1]
```diff
@@ -273,7 +273,7 @@ func filterEndpoints(ctx context.Context, writeNamespace string, kubeEndpoints [
 			}
 			return '-'
 		}, addr.Address)
-		endpointName := fmt.Sprintf("ep-%v-%v-%x", dnsname, addr.Port, hasher.Sum(nil))
+		endpointName := fmt.Sprintf("ep-%v-%v-%x", dnsname, addr.Port, hasher.Sum64())
```
Comment (EItanya) [inline projects/gloo/pkg/plugins/kubernetes/eds.go]: I don't think I made this change in mine, any specific reason for this?

**Pull request merged**